Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue2577.cs

```diff
@@ -49,5 +47,41 @@ public DetailPage()
 				Content = button;
 			}
 		}
+
+#if UITEST
+		[Test]
+		public void Issue2577Test()
+		{
+			RunningApp.WaitForElement("NavigateButton");
+			RunningApp.Tap("NavigateButton");
+
+			RunningApp.WaitForElement("ToggleBackButton");
+
+			RunningApp.Screenshot("Hamburger menu icon is visible");
+
+			AppResult[] items = RunningApp.Query("OK");
+			Assert.AreNotEqual(items.Length, 0);
+			RunningApp.Tap("OK");
+
+			RunningApp.Screenshot("Flyout menu is showing");
+
+			RunningApp.SwipeRightToLeft();
```

Comment:
Can we wait on something to ensure this action results in the desired effect? We finally got our UITests green and one thing we discovered is we had race conditions because we didn't verify with a wait after every action.. Sometimes it's not possible.. but just in case..